src/maths.f90

```diff
@@ -2153,11 +2242,233 @@ FUNCTION COTH_DP(A)
     RETURN
   END FUNCTION COTH_DP
 
+  !
+  !================================================================================================================================
+  !
+
+  !> Calculates second derivatives of a cubic spline function for a tabulated function y(x). Call  spline_cubic_val to evaluate at t values.
```

Comment:
Hi @chrispbradley 

I'm just wondering now if it is OK to use these spline interpolation routines like this or if the full source needs to be incorporated into OpenCMISSExtras. It's distributed under the LGPL. 

Cheers